pages/workspace/workspace.html.js

```diff
@@ -288,11 +312,28 @@ function Workspace({ navTitle, showLogs, resLocals }) {
 
             socket.emit('startWorkspace', { workspace_id: workspaceId });
 
-            setInterval(() => {
+            const intervalId = setInterval(() => {
               socket.emit('heartbeat', { workspace_id: workspaceId }, (msg) => {
                 console.log('heartbeat, msg =', msg);
               });
             }, heartbeatInterval * 1000);
+
+            let timeoutId = null;
+            document.addEventListener('visibilitychange', () => {
+              if (document.visibilityState == 'hidden') {
+                timeoutId = setTimeout(() => {
+                  clearInterval(intervalId);
+                  workspaceFrame.src = 'abount:blank';
+                  showStoppedFrame();
+                }, 10 * 1000);
```

Comment:
This is set artificially low for testing. This should probably be configurable too.

Reply:
Made configurable in 2323f70. In theory, this is something that we could make vary per course/institution to allow for different "tiers" of service in the future.